drivers/gpio/gpio_cy8c95xx.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2021 Synopsys
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT cypress_cy8c95xx_gpio_port
+
+#include <errno.h>
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/gpio.h>
+#include <drivers/i2c.h>
+#include <sys/byteorder.h>
+#include <sys/util.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(cy8c95xx, CONFIG_GPIO_LOG_LEVEL);
+
+#include "gpio_utils.h"
+
+/** Cache of the output configuration and data of the pins. */
+struct cy8c95xx_pin_state {
+	uint8_t dir;
+	uint8_t data;
```

Comment:
It might be worth to rename the field to `data_out`, `out_data` or similar to make it easier to notice that this holds output port values (in contrary to input port values).